Add a log in verifyDevicePath

jingxu97 opened the pull request:
Add a log during verifyDevicePath call so that we can differentiate the
operation hanging case with the path not exist case. 

```release-note
NONE
```

msau42 commented on pkg/volume/gce_pd/attacher.go:
```diff
@@ -173,6 +173,8 @@ func (attacher *gcePersistentDiskAttacher) WaitForAttach(spec *volume.Spec, devi
 				glog.Infof("Successfully found attached GCE PD %q.", pdName)
 				return path, nil
+			} else {
+				glog.V(4).Infof("could not verifying GCE PD (%q) is attached, device path is empty", pdName)
```
How frequent is this going to be?

Also "verifying" -> "verify"

jingxu97 replied:
it is every second. The timeout is 10 minutes. Actually I think 10 minutes might be a  bit of long because it will block any other operations on this device if attach fails.

msau42 replied:
Do you think every second may be too frequent and could cause log spam?

jingxu97 committed:
Add logs in GCE PD attach and mount

Add a log during verifyDevicePath call so that we can differentiate the
operation hanging case with the path not exist case.

Change-Id: I37b5166859bb99f4695a2ff188682c06f82dc13a

jingxu97 commented:
/test pull-kubernetes-integration

msau42 commented on pkg/volume/gcepd/attacher.go:
```diff
@@ -173,6 +173,8 @@ func (attacher *gcePersistentDiskAttacher) WaitForAttach(spec *volume.Spec, devi
 				glog.Infof("Successfully found attached GCE PD %q.", pdName)
 				return path, nil
+			} else {
+				glog.V(4).Infof("could not verify GCE PD (%q) is attached, device path is empty", pdName)
```
hm I wonder if every second may get too verbose.  Could we add some throttling to this so we only log once every 10 seconds?

verult replied:
Having it at log level 4 should be enough IMO. When I set log levels higher for debugging I'd like to see the log exactly when the error occurs

msau42 commented on pkg/volume/gcepd/gce_pd.go:
```diff
@@ -414,7 +414,9 @@ func (b *gcePersistentDiskMounter) SetUpAt(dir string, fsGroup *int64) error {
 	}
 
```
Can you also add a successful mount message here?  It would be nice from the logs to be able to tell how much time the bind mount takes vs how much time the fsgroup setup takes.

davidz627 commented:
/remove-lifecycle stale
would still be nice to have

jingxu97 commented:
@msau42 @verult could you help review again?

msau42 commented:
/approve

We should add similar logs to the csi driver too

verult commented:
/lgtm
